### PR TITLE
Edit viz in a dialog

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
[Describe what this change achieves]

https://github.com/user-attachments/assets/0d0560d5-d1df-43af-bdc2-f98d9ab4d3a5

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/visualization/text2viz.tsx [hunk 1]
```diff
@@ -196,42 +201,48 @@ export const Text2Viz = () => {
...
-    }
+  const onSubmit = useCallback(
+    async (inputInstruction: string = '') => {
+      setCurrentInstruction(inputInstruction);
+
+      if (status === 'RUNNING' || !selectedSource) return;
+
+      if (
+        inputQuestion.trim().length > TEXT2VEGA_INPUT_SIZE_LIMIT ||
+        inputInstruction.trim().length > TEXT2VEGA_INPUT_SIZE_LIMIT
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply public/components/visualization/text2viz.tsx:212]: I think we shouldn't, checking separately to ensure each input has its own quota

File: public/components/visualization/text2viz.scss [hunk 2]
```diff
@@ -17,10 +17,26 @@
...
-    right: 16px;
     top: 4px;
+    right: 16px;
     z-index: 9999;
+
+    // No existing button from OUI with the same style, have to customize here
+    .vizStyleEditor__editButton {
+      height: 22px;
+      padding: 2px;
+      font-size: 12px;
```
Comment (Hailong-am) [inline public/components/visualization/text2viz.scss:30]: do we need to customize the font size? can use `$ouiFontSizeXS` ?

File: public/components/visualization/text2viz.tsx [hunk 3]
```diff
@@ -174,7 +178,8 @@ export const Text2Viz = () => {
             }
           }
           if (savedVis?.uiState) {
-            setInput(JSON.parse(savedVis.uiState ?? '{}').input);
+            setInputQuestion(JSON.parse(savedVis.uiState ?? '{}').input ?? '');
+            setCurrentInstruction(JSON.parse(savedVis.uiState ?? '{}').instruction ?? '');
```
Comment (Hailong-am) [inline public/components/visualization/text2viz.tsx:182]: do we need to add try catch here in case uiState is invalid json?

**Review by Hailong-am — APPROVED** [review]

File: public/components/visualization/text2viz.tsx (hunk 3 above)
Comment (ruanyl) [reply public/components/visualization/text2viz.tsx:182]: The exception should be handled by the `.catch`

File: public/components/visualization/text2viz.scss (hunk 2 above)
Comment (ruanyl) [reply public/components/visualization/text2viz.scss:30]: good point, let me check that.

**Pull request merged**